pkg/frontend/session.go

```diff
@@ -637,12 +637,12 @@ func (th *TxnHandler) CommitTxn() error {
 	if !th.IsValidTxn() {
 		return nil
 	}
-	var ctx context.Context
-	if th.ses != nil {
-		ctx = th.ses.GetRequestContext()
-	} else {
-		ctx = context.Background()
+	ctx := th.ses.GetRequestContext()
+	if ctx == nil {
+		panic("context should not be nil")
 	}
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
```

Comment:
Provide a flexible method to change the value of the timeout instead of hard code.